ci-operator/config/openshift/coredns-ocp-dnsnameresolver/openshift-coredns-ocp-dnsnameresolver-main.yaml

```diff
@@ -7,7 +7,7 @@ build_root:
   image_stream_tag:
     name: release
     namespace: openshift
-    tag: golang-1.21
+    tag: rhel-9-release-golang-1.22-openshift-4.17
```

Comment:
I think that we can set up the build_root from the repository [like it's done in CoreDNS](https://github.com/openshift/release/blob/776cf6df9571b2422bfd9f355f50e5de92fe4722/ci-operator/config/openshift/coredns/openshift-coredns-master.yaml#L14-L15). This would simplify the ci config and move the version update to the repository where this decision is taken.